posts/kotlin-like-java-but-better-2.md

```diff
@@ -0,0 +1,255 @@
+```
+
+### /api/people/add
+We can then add a new person to the service using the `/people/add` endpoint. Feel free to edit this command to add several people! We expect the first and last name of the person we added to be returned.
+```
+curl --request POST \
+  --url http://localhost:8080/api/people/add \
+  --header 'Content-Type: application/json' \
+  --data '{
+	"firstName": "Jane",
+	"lastName": "Doe",
+	"age": 32
+}'
+```
+The expected response looks like:
+```
+Jane Doe
+```
+
+### /api/people
+Finally, we can request the list of people with the `/people` endpoint.
+```
+curl --request GET \
+  --url http://localhost:8080/api/people
+```
+We expect a response that looks something like this (assuming you've added two People so far):
+```
+[
+  {
+    "firstName": "Jane",
+    "lastName": "Doe",
+    "age": 32
+  },
+  {
+    "firstName": "John",
+    "lastName": "Doe",
+    "age": 34
+  }
+]
+```
+As we continue working with the service, you may want to use a tool like [Postman](https://www.postman.com/) or [Insomnia](https://insomnia.rest/) instead of manually editing the curl commands.
+
+# Diving Into the Code
+Now that we have explored the general behavior of the application, let's take a look at the `Person.java`, `PersonService.java`, and `PersonController.java` classes and convert them to Kotlin classes.
+
+There is also a test suite included with this repo that we will be using to ensure that the behavior and functionality of the application does not change as we convert our application.
+## The `Person` Class
+### Person.java
+```Java
+public class Person {
+    private String firstName;
+    private String lastName;
+    private int age;
+
+    public Person(String firstName, String lastName, int age) {
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.age = age;
+    }
+
+    public String getFirstName() {
+        return this.firstName;
+    }
+
+    public String getLastName() {
+        return this.lastName;
+    }
+
+    public int getAge() {
+        return this.age;
+    }
+
+    public void setFirstName(String name) {
+        this.firstName = name;
+    }
+
+    public void setLastName(String name) {
+        this.lastName = name;
+    }
+
+    public void setAge(int age) {
+        this.age = age;
+    }
+
+    public String toString() {
+        return this.firstName + " " + this.lastName;
+    }
+}
+```
+This class represents the Person model that our service works with. This is a fairly basic Plain Old Java Object (POJO) that includes several of boilerplate getter and setter functions. As we are rewriting this class in Kotlin, this is a great opportunity to utilize [Data Classes](https://kotlinlang.org/docs/data-classes.html), which already have default getter and setter functions, as well as defaults for other common functions such as `toString()`. We can delete the old Person.Java file, and create a new Person.kt Kotlin file that looks like this:
```

Comment:
This is a fairly basic Plain Old Java Object (POJO) that includes several boilerplate getter and setter functions.